src/main/java/com/entropyteam/entropay/employees/services/PtoService.java

```diff
@@ -192,4 +193,20 @@ private CalendarEventDto createCalendarEventDto(Pto pto){
         String eventName = employee.getFirstName() + " " + employee.getLastName() + " " + leaveType.getName();
         return new CalendarEventDto(eventId, eventName, startDate, endDate);
     }
+
+    public PtoDto cancelPto(UUID id, PtoDto ptoDto) {
+        Pto oldEntity = ptoRepository.findById(id).orElseThrow();
+        Pto entityToUpdate = toEntity(ptoDto);
+        entityToUpdate.setId(id);
+        entityToUpdate.setStatus(entityToUpdate.getStatus());
```

Comment:
for the cancellation you just need to set status CANCELLED to the oldEntity (probably a better name):
`entityToUpdate.setStatus(Status.CANCELLED);`

Reply:
Resolve